pkg/virt-handler/vm.go

```diff
@@ -547,6 +557,42 @@ func (c *VirtualMachineController) Execute() bool {
 	return true
 }
 
+func (d *VirtualMachineController) getMigrationConfig(vmi *v1.VirtualMachineInstance) (migrationConfig *v1.MigrationConfig) {
```

Comment:
i think we should avoid caching the migration objects in virt-handler.  there's actually an opportunity in the migration controller for us to copy the config to the VM's status.  

take a look at this code block below in pkg/virt-controller/watch/migration.go.  We can copy the migration config to the vmi.status.migrationstate struct and virt-handler can get it that way. 

```
case virtv1.MigrationScheduled:
                // once target pod is scheduled, alert the VMI of the migration by
                // setting the target and source nodes. This kicks off the preparation stage.
                if podExists && !podIsDown(pod) {
                        vmiCopy := vmi.DeepCopy()
                        vmiCopy.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{                   
                                MigrationUID: migration.UID,
                                TargetNode:   pod.Spec.NodeName,  
                                SourceNode:   vmi.Status.NodeName,
                                TargetPod:    pod.Name,
                        }

                        // By setting this label, virt-handler on the target node will receive
                        // the vmi and prepare the local environment for the migration
                        vmiCopy.ObjectMeta.Labels[virtv1.MigrationTargetNodeNameLabel] = pod.Spec.NodeName              

                        if !reflect.DeepEqual(vmi.Status, vmiCopy.Status) ||
                                !reflect.DeepEqual(vmi.Labels, vmiCopy.Labels) {
                                _, err := c.clientset.VirtualMachineInstance(vmi.Namespace).Update(vmiCopy)             
                                if err != nil {             
                                        c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedHandOverPodReason, fmt.Sprintf("Failed to set MigrationStat in VMI status. :%v", err))
                                        return err
                                }
                                c.recorder.Eventf(migration, k8sv1.EventTypeNormal, SuccessfulHandOverPodReason, "Migration target pod is ready for preparation by virt-handler.")
                        } 
                }
        }

```

Reply:
Ok, sounds good. I didn't do it initially because I was trying to avoid introducing additional fields into the migration status, but this definitely removes the need for querying anything.

Reply:
Done